Marlin/gcode.h

```diff
@@ -148,8 +147,7 @@ class GCodeParser {
 
     // Code is found in the string. If not found, value_ptr is unchanged.
     // This allows "if (seen('A')||seen('B'))" to use the last-found value.
-    // This is volatile because its side-effects are important
-    static volatile bool seen(const char c) {
+    static bool seen(const char& c) {
```

Comment:
The purpose of `volatile` here is to ensure that every time you use `parser.seen()` a real function call is happening, and the compiler is not substituting it with the results of the previous call. What is the purpose of removing `volatile`?